tests/proxyMap.test.tsx

```diff
@@ -375,3 +376,40 @@ describe('snapshot', () => {
     expect(snap2.get('key1')).toBe('val1modified')
   })
 })
+
+describe('ui updates - useSnapshot', async () => {
+  it('should update ui when calling has before and after deleting a key', async () => {
+    const state = proxyMap()
+    const TestComponent = () => {
+      const snap = useSnapshot(state)
+
+      return (
+        <>
+          <p>has key: {`${snap.has('key')}`}</p>
+          <button onClick={() => state.set('key', 'value')}>set key</button>
```

Comment:
Try something like this:
```suggestion
          <button onClick={() => { state.set('key', 'value'); state.set('key2', 'value'); }}>set key</button>
```

Reply:
This is a good reason 😅😬

Reply:
I added an epoch to keep track of versions instead of using this.data.length and added tests to test for adding and deleting multiple keys as well as clearing the map/set